src/strings_utils.cpp

```diff
@@ -232,7 +232,16 @@ std::ostream & operator<<(std::ostream & os, boost::any & obj) {
 }
 
 bool is_ascii_normal(const std::string str){
-	auto it = find_if(str.begin(), str.end(), [](auto x){return !isprint(x);	} );
+	auto it = find_if(str.begin(), str.end(), [](int x) {
+		// The behavior is undefined if the value of ch is not representable as unsigned char and is not equal to EOF.
+		// http://en.cppreference.com/w/cpp/string/byte/isprint
+		if ((x < std::numeric_limits<unsigned char>::min()
+			|| x > std::numeric_limits<unsigned char>::max())
+			&& x != EOF)
+				return true;
```

Comment:
bool in_range =   ( x < ... )  &&  ( x > ...) ;
bool valid = in_range && ( x!= EOF );
if ( ! valid ) return true;

would be more obvious to read